pkg/hoist/hoist_launchable.go

```diff
@@ -239,30 +233,12 @@ func (hl *Launchable) Install() error {
 		return nil
 	}
 
-	outDir, err := ioutil.TempDir("", hl.Version())
-	defer os.RemoveAll(outDir)
-	if err != nil {
-		return util.Errorf("Could not create temporary directory to install %s: %s", hl.Version(), err)
-	}
-
-	outPath := filepath.Join(outDir, hl.Version())
-
-	err = hl.FetchToFile(hl.Location, outPath)
+	data, err := hl.Fetcher.Open(hl.Location)
```

Comment:
I definitely agree with you.  There should be some strategy for caching artifacts, not just for reuse by different pods but also for resilience in case we need to undo a rollout and go back to using an earlier version of a pod.

That said, I don't want to address that in this diff.  This diff is a refactor of existing behavior.  And I suspect that writing an artifact cache will be enough extra work that this simplification won't really get in the way for now.